fix(linter): skip eslint/no-redeclare when running on modules

camc314 opened the pull request:
eslint rule explicitly specifies `sourceType: "script"`, as `modules` execute with their own global scope

https://github.com/eslint/eslint/blob/129882d2fdb4e7f597ed78eeadd86377f3d6b078/tests/lib/rules/no-redeclare.js#L22

We can skip running this rule on moudle code ( i tested eslint's tests with `sourceType: module`, and non of the test cases throw errors

closes #10912

camc314 commented:
* **#11004** <a href="https://app.graphite.dev/github/pr/oxc-project/oxc/11004?utm_source=stack-comment-icon" target="_blank"><img src="https://static.graphite.dev/graphite-32x32-black.png" alt="Graphite" width="10px" height="10px"/></a> 👈 <a href="https://app.graphite.dev/github/pr/oxc-project/oxc/11004?utm_source=stack-comment-view-in-graphite" target="_blank">(View in Graphite)</a>
* `main`
<hr />

## How to use the Graphite Merge Queue

Add either label to this PR to merge it via the merge queue:
- <kbd>0-merge</kbd> - adds this PR to the back of the merge queue
- <kbd>hotfix</kbd> - for urgent hot fixes, skip the queue and merge this PR next

You must have a Graphite account in order to use the merge queue. Sign up using [this link](https://app.graphite.dev/invite/github/oxc-project?ref=merge-queue-instructions-in-stack-comment&prId=11004).

<h2></h2>
<sub>

An organization admin has enabled the [Graphite Merge Queue](https://graphite.dev/docs/graphite-merge-queue) in this repository.

</sub>

<sub>

Please do not merge from GitHub as this will restart CI on PRs being processed by the merge queue.

</sub>

<h2></h2>

This stack of pull requests is managed by <a href="https://graphite.dev?utm-source=stack-comment"><b>Graphite</b></a>. Learn more about <a href="https://stacking.dev/?utm_source=stack-comment">stacking</a>.
<!-- Current dependencies on/for this PR: -->

camc314 commented:
yeah i believe this change now follows that spec

`ModuleKind` comes from here, which is retrieved from the file extension
https://github.com/oxc-project/oxc/blob/c50b1a564dac959506d4101975f3213ab7670a24/crates/oxc_span/src/source_type/mod.rs#L145

Boshen commented:
### Merge activity

* **May 14, 12:45 AM EDT**: The merge label '0-merge' was detected. This PR will be added to the [Graphite merge queue](https://app.graphite.dev/merges?org=oxc-project&repo=oxc) once it meets the requirements.
* **May 14, 12:45 AM EDT**: `Boshen` added this pull request to the [Graphite merge queue](https://app.graphite.dev/merges?org=oxc-project&repo=oxc).

camc314 committed:
fix(linter): skip eslint/no-redeclare when running on modules (#11004)

eslint rule explicitly specifies `sourceType: "script"`, as `modules` execute with their own global scope

https://github.com/eslint/eslint/blob/129882d2fdb4e7f597ed78eeadd86377f3d6b078/tests/lib/rules/no-redeclare.js#L22

We can skip running this rule on moudle code ( i tested eslint's tests with `sourceType: module`, and non of the test cases throw errors

closes #10912
